### PR TITLE
Use new ExecutionStatus: [ECR-4222]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Migrate the light client to use the extended ExecutionStatus,
currently available from ejb-common module.

The ExecutionStatus, as previously, is exposed as a protobuf
class as a simpler approach compared to a separate
user-facing class (which would also require all the documentation
already present in the message).

---
See: https://jira.bf.local/browse/ECR-4222

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java [hunk 1]
```diff
@@ -205,38 +212,97 @@ void parseServicesResponse() {
...
-        arguments(ExecutionStatuses.success(), successStatus),
-        arguments(ExecutionStatuses.serviceError(errorCode, errorDescription), serviceErrorStatus),
-        arguments(ExplorerApiHelper.buildExecutionStatus(ErrorKind.DISPATCHER, errorCode,
-            errorDescription),
-            dispatcherErrorStatus),
-        arguments(ExplorerApiHelper.buildExecutionStatus(ErrorKind.RUNTIME, errorCode,
-            errorDescription),
-            runtimeErrorStatus),
-        arguments(ExplorerApiHelper.buildPanicExecutionStatus(errorDescription), panicStatus)
+    List<StatusParameters> combinations = ImmutableList.of(
```
Comment (dmitry-timofeev) [inline exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: Alternatively, we could test _separately_:
* All the callSite types.
* All the status types (with same call site).

File: exonum-light-client/src/main/java/com/exonum/client/ExplorerApiHelper.java [hunk 2]
```diff
@@ -169,36 +140,143 @@ static ExecutionStatus buildExecutionStatus(ErrorKind errorKind, int code,
...
-    GetTxResponseExecutionResult status;
+    JsonObject locationProof; // TODO: in scope of LC P3
+    GetTxResponseExecutionStatus status;
   }
 
   /**
    * Json object wrapper for transaction execution result, i.e.,
    * {@code "$.status"}.
+   *
+   * <p>See serde::ExecutionStatus: https://github.com/exonum/exonum/blob/v1.0.0-rc.1/exonum/src/runtime/error/execution_status.rs#L102
```
Comment (vitvakatu) [inline exonum-light-client/src/main/java/com/exonum/client/ExplorerApiHelper.java]: What do you mean by `serde` prefix?

File: exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java [hunk 3]
```diff
@@ -205,38 +212,97 @@ void parseServicesResponse() {
...
+    for (StatusParameters params: combinations) {
+      arguments.add(
+          arguments(
+              ExecutionStatus.newBuilder()
+                  .setError(
+                      ExecutionError.newBuilder(errorTemplate)
+                          .setKind(params.errorKind)
+                          .setCallSite(
+                              CallSite.newBuilder(callSiteTemplate)
+                                  .setCallType(params.callSiteType)))
```
Comment (vitvakatu) [inline exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: 🎢 😀

File: exonum-light-client/src/main/java/com/exonum/client/ExplorerApiHelper.java (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-light-client/src/main/java/com/exonum/client/ExplorerApiHelper.java]: The module: there are two ExecutionStatuses, the one which is used for JSON serialization is in module serde

**Pull request merged**